### PR TITLE
Migrate Vue3 to use `vite` instead of deprecated `vue-cli`

### LINKED ISSUE #57: Jest unit test support for Vue
Add Jest support and write one or two thoughtful tests as examples.
Should run automatically for all PRs.
These tests are for the generated project, but should be run on bootstrapper PRs as well to validate that new bootstrapper changes don't break functionality in the generated apps.

### PR DESCRIPTION
**Pull request by lakardion**
## What this does

Replace `vue-cli` with `vite`. This is the recommended way of starting new vue projects.
`vue-cli` is in maintenance mode and won't receive any more updates.

- As part of the upgrade, there's also added support for typescript. Although most files are `js` right now, the configuration and compilation step is ready for taking in `ts` files.

Closes #57 

## Checklist
- [x] Make cypress work
- [x] Move unit tests to vitest

## How to test

Add user steps to achieve desired functionality for this feature.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/clients/web/vue3/package.json [hunk 1]
```diff
@@ -1,55 +1,44 @@
...
-  "description": "{{ cookiecutter.project_name }} Client Application",
+  "name": "vuevite",
   "private": true,
+  "version": "0.0.0",
+  "type": "module",
   "scripts": {
-    "serve": "vue-cli-service serve",
-    "build": "vue-cli-service build",
-    "cypress": "source ../.env && npx cypress open",
-    "test:unit": "vue-cli-service test:unit",
```
Comment (lakardion) [inline {{cookiecutter.project_slug}}/clients/web/vue3/package.json:10]: Will get these back at some point in this PR

**Comment by lakardion** [comment]
Should be good to review!

File: {{cookiecutter.project_slug}}/clients/web/vue3/package.json [hunk 2]
```diff
@@ -3,53 +3,50 @@
...
+    "js-cookie": "3.0.5",
     "vue": "^3.3.4",
-    "vue-router": "^4.2.4",
-    "vuex": "^4.1.0",
-    "vuex-persistedstate": "^4.1.0",
-    "zod": "3.22.2"
+    "vue-router": "4.2.4",
+    "vuex": "4.1.0",
+    "vuex-persistedstate": "4.1.0",
+    "zod": "3.21.4"
```
Comment (mayakeeley) [inline {{cookiecutter.project_slug}}/clients/web/vue3/package.json:30]: @lakardion I know this is a tiny change but how come we're going down a version here?
Similarly with vuex changing from ^4.1.0 to 4.1.0

File: {{cookiecutter.project_slug}}/clients/web/vue3/src/main.ts [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import { createApp } from 'vue'
+import './main.css'
 import App from './App.vue'
-import router from './router'
+//TODO:
```
Comment (mayakeeley) [inline {{cookiecutter.project_slug}}/clients/web/vue3/src/main.ts:4]: Should: remove TODO

File: {{cookiecutter.project_slug}}/clients/web/vue3/tests/e2e/specs/test-login.cy.ts [hunk 4]
```diff
@@ -6,6 +6,7 @@ describe('Tests login workflow', () => {
   })
 
   it('Filling in email and passwords goes to', () => {
+    cy.visit('/login')
```
Comment (mayakeeley) [inline {{cookiecutter.project_slug}}/clients/web/vue3/tests/e2e/specs/test-login.cy.ts:9]: Has this not happened in the previous it statement?

File: {{cookiecutter.project_slug}}/clients/web/vue3/public/vite.svg
Comment (mayakeeley) [inline {{cookiecutter.project_slug}}/clients/web/vue3/public/vite.svg:1]: Question: is this file needed since we don't use it for favicons?

File: {{cookiecutter.project_slug}}/clients/web/vue3/tests/e2e/specs/test-login.cy.ts (hunk 4 above)
Comment (lakardion) [reply {{cookiecutter.project_slug}}/clients/web/vue3/tests/e2e/specs/test-login.cy.ts:9]: I think cypress changed the way they run the test. They clean up in between. It starts back at home

Comment (lakardion) [reply {{cookiecutter.project_slug}}/clients/web/vue3/tests/e2e/specs/test-login.cy.ts:9]: (This upgraded cypress from 8 -> 13 )

Comment (mayakeeley) [reply {{cookiecutter.project_slug}}/clients/web/vue3/tests/e2e/specs/test-login.cy.ts:9]: Ah great, that makes sense

File: {{cookiecutter.project_slug}}/clients/web/vue3/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    "src/**/*.tsx",
+    "src/**/*.vue",
+    "tests/**/*.ts"
+  ],
+  "references": [
+    {
+      "path": "./tsconfig.node.json"
+    }
+  ]
+}
```
Comment (paribaker) [inline {{cookiecutter.project_slug}}/clients/web/vue3/tsconfig.json:48]: 👏

File: {{cookiecutter.project_slug}}/clients/web/vue3/package.json (hunk 2 above)
Comment (lakardion) [reply {{cookiecutter.project_slug}}/clients/web/vue3/package.json:30]: @mayakeeley

**Review by mayakeeley — APPROVED** [review]
lgtm!
